src/Orchard.Web/Modules/Orchard.Users/Controllers/AccountController.cs

```diff
@@ -224,27 +236,76 @@ public class AccountController : Controller {
         [SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes",
             Justification = "Exceptions result in password not being changed.")]
         public ActionResult ChangePassword(string currentPassword, string newPassword, string confirmPassword) {
-            ViewData["PasswordLength"] = MinPasswordLength;
+            var membershipSettings = _membershipService.GetSettings();
+            ViewData["PasswordLength"] = membershipSettings.GetMinimumPasswordLength(); ;
 
-            if ( !ValidateChangePassword(currentPassword, newPassword, confirmPassword) ) {
+            if (!ValidateChangePassword(currentPassword, newPassword, confirmPassword)) {
                 return View();
             }
 
+            if (PasswordChangeIsSuccess(currentPassword, newPassword, _orchardServices.WorkContext.CurrentUser.UserName)) {
+                return RedirectToAction("ChangePasswordSuccess");
+            }
+            else {
+                return ChangePassword();
+            }
+        }
+
+        [AlwaysAccessible]
+        public ActionResult ChangeExpiredPassword(string username) {
```

Comment:
Do we really need a new way to change the password? Why not use the default action/view? I don't see any real value of testing the validity again here.